### ISSUE TITLE
Switch site to GitHub pages?

### ISSUE THREAD

**Issue by ctrueden** [vrapper/vrapper#673]
With the extended SourceForge downtime, and SourceForge generally going down a lot, would it make sense to switch the Vrapper website and update sites over to GitHub pages?

**Comment by keforbes** [comment]
I agree, I've had intermittent issues with SourceForge for a while now.  I'd love to move everything over to GitHub.  Can GitHub Pages host arbitrary files though?  I don't think we'd be able to create an Eclipse update site from the GitHub release artifacts.  Right now I copy those files into SourceForge's web server to act as the update site.

I need to learn more about the limitations of GitHub Pages to know if it'd be viable.

**Comment by keforbes** [comment]
Is this something that I as the owner of the project would have to setup or can someone create a pull request for this?  I'd love some help making a new website if someone is available.

**Comment by albertdev** [comment]
I wouldn't call Github a real replacement.

* They don't do dynamic pages (PHP, Python, ...) so the site would have to be rewritten.
* They don't use a dedicated storage space, instead they want you to push files into a branch of your main repo, increasing it's size.

**Comment by ctrueden** [comment]
@albertdev wrote:
> They don't use a dedicated storage space, instead they want you to push files into a branch of your main repo, increasing it's size.

The organization site is a separate repo though: `vrapper.github.io`. The master branch of that repo would map to `https://vrapper.github.io/` which would avoid bloating the size of the actual `vrapper` code repository.

Agreed about the static pages though: dynamic sites (e.g., driven by CMS) will not work with GitHub Pages. But from my naive understanding of Eclipse update sites, they could work.

**Comment by ctrueden** [comment]
One quick web search later, I guess the update site shouldn't live on GitHub Pages:
    http://stackoverflow.com/a/5263884

But Bintray could be good for that!
    http://blog.bintray.com/2014/03/11/hosting-your-eclipse-update-site-p2-on-bintray-com/

**Comment by ctrueden** [comment]
@keforbes:
> can someone create a pull request for this?

Not exactly a PR, but someone could create a new vrapper.github.io in their own user space, to show you what it would look like, then transfer to you if you like it. I am very sorry to say I cannot work on it in the next few weeks personally. It would be a really fun project, but my summer is totally booked. So if no one else takes care of it, perhaps this fall...

**Comment by albertdev** [comment]
Looks like our home page and most likely the update site is back for now...

**Comment by hrj** [comment]
:+1: for hosting on GH pages. Also, the file quotas are [generous](https://help.github.com/articles/what-is-my-disk-quota/) enough: 100MB files are allowed, while the vrapper jars are in KBs. Also see [this answer](https://www.quora.com/What-are-bandwidth-and-traffic-limits-for-GitHub-pages?share=1) from a githubber.

----

I can help with creating a basic site to experiment with. The first steps are quite easy.

You just need to create a repo named `vrapper.github.io` under the `vrapper` organisation, clone the repo locally, add your files and push. The files will be automatically served from `https://vrapper.github.io`.

Let me know if you need any help.

Would love to give back to this project in whatever simple way I can.

**Comment by keforbes** [comment]
I think I've already created vrapper.github.io as a simple redirect to our SourceForge page.  Can you create a pull request on that branch?  I'd appreciate any help you can give to setup a new webpage for us.

**Comment by keforbes** [comment]
Sorry, I meant we already have a `gh-pages` branch.  Is that the old way of doing things?  It seems to still work.

**Comment by hrj** [comment]
Yeah, gh-pages is the old way, and yeah it still works. But I only see a repo called `vrapper.github.com` and it only has a `master` branch. Do you have another private repo that is not visible to public?

I think PRs are over kill for this. Either you can make me a contributor for that repo or if you don't wish to do that (perfectly understandable) you can just commit the `vrapper-stable-update-site` into the `gh-pages` branch.

Github will automatically serve the new version after you push to the branch.

After that, the update site should be available on `https://vrapper.github.io/vrapper-stable-update-site`

There might be some configuration tweaks required, but we wouldn't know until we try it out.

**Comment by keforbes** [comment]
Awesome, thanks, I'll try setting that up sometime this weekend.

**Comment by albertdev** [comment]
@keforbes 
The gh-pages branch ends up at http://vrapper.github.io/vrapper/

The [vrapper.io.github](https://github.com/vrapper/vrapper.github.com) repo is what actually provides http://vrapper.github.io/.

Personally I would put it in the vrapper.github.io repo so that I or others don't have to pull in update sites each time one does `git fetch --all`.

**Comment by keforbes** [comment]
Thanks for the info guys, I should be able to clean all this up this weekend.  I'll get rid of gh-pages, vrapper.github.com and create a vrapper.github.io.  

It looks like GitHub Pages supports Jekyll for templating.  @hrj, would you be able to help me replace our php with Jekyll markdown?  We're only using php to do includes anyway.

**Comment by keforbes** [comment]
Ok, https://github.com/vrapper/vrapper.github.io has been created and I gave write access to @albertdev  and @hrj.  I moved the entire `website/` directory into that new repo as a starting point but we can change the website however we want.

I created a dummy index.html file just to test things out.  GitHub Pages refused to load index.php but I think we can replace all of our php with markdown anyway so that shouldn't be a problem.

**Comment by hrj** [comment]
Cool. I have created some issues here: https://github.com/vrapper/vrapper.github.io/issues

Will first try to get the update-site hosted on GH Pages [#1](https://github.com/vrapper/vrapper.github.io/issues/1) and then look at the conversion of php files [#2](https://github.com/vrapper/vrapper.github.io/issues/2).

**Comment by hrj** [comment]
I have added the stable site to GH pages. It can be accessed from: https://vrapper.github.io/update-site/vrapper-stable/

Please test it in Eclipse and give me feedback.
